Tests time out on slow machines

The test suite is run in [citgm](https://github.com/nodejs/citgm) as smoke test to verify that Node.js changes do not break this or other libraries.

We do encounter timeouts though from time to time. It would be great if the dedicated tests either get an increased timeout value or get refactored in a way that pass faster.

See e.g, https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/2142/nodes=debian9-64/testReport/junit/(root)/citgm/uglify_js_v3_7_2/ (available ~7 days)

@alexlamsl sorry, I forgot about that.

Here the output (the color detection is not working correct in the terminal, so please ignore the ANSI codes):

```
 [31m3 test(s) failed![39m
 bin/uglifyjs
   1) Should compress swarm of unused variables with reasonable performance
 [31mError: Timed out: exceeds 5000ms[39m
     at Timeout.<anonymous> (/home/iojs/tmp/citgm_tmp/43778314-1e60-4261-a3e2-21022c837735/uglify-js/test/mocha.js:71:27)
     at listOnTimeout (internal/timers.js:537:17)
     at processTimers (internal/timers.js:481:7)
 bin/uglifyjs with input file globs
   2) bin/uglifyjs with multiple input file globs.
 [31mError: Timed out: 6392ms > 5000ms[39m
     at done (/home/iojs/tmp/citgm_tmp/43778314-1e60-4261-a3e2-21022c837735/uglify-js/test/mocha.js:61:23)
     at /home/iojs/tmp/citgm_tmp/43778314-1e60-4261-a3e2-21022c837735/uglify-js/test/mocha/cli.js:746:13
     at ChildProcess.exithandler (child_process.js:292:7)
     at ChildProcess.emit (events.js:304:20)
     at maybeClose (internal/child_process.js:1028:16)
     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
   3) Should handle array of glob strings - matching and otherwise
 [31mError: Timed out: exceeds 5000ms[39m
     at Timeout.<anonymous> (/home/iojs/tmp/citgm_tmp/43778314-1e60-4261-a3e2-21022c837735/uglify-js/test/mocha.js:71:27)
     at listOnTimeout (internal/timers.js:537:17)
     at processTimers (internal/timers.js:481:7)
 npm ERR! Test failed.  See above for more details.
```

--- a/test/mocha.js
+++ b/test/mocha.js
@@ -1,7 +1,7 @@
 var fs = require("fs");
 
 var config = {
-    limit: 5000,
+    limit: 10000,
     timeout: function(limit) {
         this.limit = limit;
     }
@@ -55,11 +55,11 @@ process.nextTick(function run() {
         var elapsed = Date.now();
         var timer;
         var done = function() {
-            reset();
             elapsed = Date.now() - elapsed;
             if (elapsed > task.limit) {
                 throw new Error("Timed out: " + elapsed + "ms > " + task.limit + "ms");
             }
+            reset();
             log_titles(console.log, task.titles, green('\u221A '));
             process.nextTick(run);
         };

